pkg/controllers/node/termination/controller.go

```diff
@@ -123,14 +125,14 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 }
 
 func (c *Controller) deleteAllNodeClaims(ctx context.Context, node *v1.Node) error {
-	nodeClaimList := &v1beta1.NodeClaimList{}
-	if err := c.kubeClient.List(ctx, nodeClaimList, client.MatchingFields{"status.providerID": node.Spec.ProviderID}); err != nil {
+	nodeClaimList, err := nodeutils.GetNodeClaims(ctx, node, c.kubeClient)
```

Comment:
```suggestion
	nodeClaims, err := nodeutils.GetNodeClaims(ctx, node, c.kubeClient)
```

nit: May not need to call it a list here since it's no longer an actual `NodeClaimList` representation